operator/src/main/java/org/keycloak/operator/crds/v2alpha1/deployment/IngressSpec.java

```diff
@@ -0,0 +1,19 @@
+package org.keycloak.operator.crds.v2alpha1.deployment;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import io.sundr.builder.annotations.Buildable;
+
+@Buildable(editableEnabled = false, builderPackage = "io.fabric8.kubernetes.api.builder")
+public class IngressSpec {
+
+    @JsonProperty("enabled")
+    private boolean ingress;
```

Comment:
Actually I had it implemented like that initially but I was not sure if it's fine to assign a default value to it. Thanks for pointing it out, Martin.

Reply:
@mabartos I introduced the new `ingressSpecDefined` parameter, based on which I can easily control, whether the spec is defined or not. Based on that I added two new test-cases. Please, could you check it once more?

> we don't have to check the "nullability" of the property later.

I think that this might not be accurate - I am rather checking the nullability of ingress spec and not its boolean property. So I would say the check is necessary one way or another (as I mentioned earlier - I had it implemented like that, so I kinda experimented with it). 
But I might be wrong or missing something that you see. Let me know. Thanks!